### PR TITLE
Fixed make_voxceleb1.pl to deal with new data format

### PR DESCRIPTION
**Pull request by sayint-ai**
The new voxceleb1 dataset follows a similar structure as voxceleb2. Nivedita has made appropriate changes to the make_voxceleb1.pl script to take care of this.

### CODE REVIEW

**Comment by danpovey** [comment]
@david-ryan-snyder  please check

**Comment by david-ryan-snyder** [comment]
Thanks. This needed to be done.

Let's keep both versions of the script. There are still a lot of people (maybe most of us) who downloaded the corpus when it first came out, and so we want to make sure the old format is still supported. Rather than replacing the existing script with the new one, could you rename the new one make_voxceleb1_v2.pl? Then, in the run.sh, you can comment out the old version, and write a comment that explains the situation (e.g., if you downloaded the corpus recently, use make_voxceleb1_v2.pl, otherwise use make_voxceleb1.pl)?

File: egs/voxceleb/v2/run.sh [hunk 1]
```diff
@@ -16,20 +16,30 @@ vaddir=`pwd`/mfcc
 
 
 # The trials file is downloaded by local/make_voxceleb1.pl.
-voxceleb1_trials=data/voxceleb1_test/trials
-voxceleb1_root=/export/corpora/VoxCeleb1
-voxceleb2_root=/export/corpora/VoxCeleb2
+#voxceleb1_trials=data/voxceleb1_test/trials
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v2/run.sh]: Why are the trials commented out? Mistake?

File: egs/voxceleb/v2/run.sh [hunk 2]
```diff
@@ -16,20 +16,30 @@ vaddir=`pwd`/mfcc
...
+voxceleb1_root=/home/data2/VoxCeleb/nivedita/VoxCeleb1
+voxceleb2_root=/home/data2/VoxCeleb/nivedita/VoxCeleb2
 nnet_dir=exp/xvector_nnet_1a
-musan_root=/export/corpora/JHU/musan
+musan_root=/home/data2/VoxCeleb/nivedita/musan
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v2/run.sh]: Would you mind changing this back to the old paths? E.g., 

voxceleb1_root=/export/corpora/VoxCeleb1
voxceleb2_root=/export/corpora/VoxCeleb2
musan_root=/export/corpora/JHU/musan

File: egs/voxceleb/v2/run.sh [hunk 3]
```diff
@@ -16,20 +16,30 @@ vaddir=`pwd`/mfcc
...
 
 if [ $stage -le 0 ]; then
   local/make_voxceleb2.pl $voxceleb2_root dev data/voxceleb2_train
   local/make_voxceleb2.pl $voxceleb2_root test data/voxceleb2_test
   # This script creates data/voxceleb1_test and data/voxceleb1_train.
   # Our evaluation set is the test portion of VoxCeleb1.
-  local/make_voxceleb1.pl $voxceleb1_root data
+  #local/make_voxceleb1.pl $voxceleb1_root data
+  local/make_voxceleb1.pl $voxceleb1_root test data/voxceleb1_test
+  local/make_voxceleb1.pl $voxceleb1_root dev data/voxceleb1_train
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v2/run.sh]: After you change the new version of the script to local/make_voxceleb1_v2.pl, please put a comment here saying that the v2 script works with the newest version of the corpus, and that if you downloaded the corpus awhile ago, it's possible you will need to use the local/make_voxceleb1.pl version.

File: egs/voxceleb/v1/local/make_voxceleb1.pl [hunk 4]
```diff
@@ -1,130 +1,127 @@
 #!/usr/bin/perl
 #
-# Copyright 2018  Ewald Enzinger
-#           2018  David Snyder
+# Copied from make_voxceleb2.pl and made necessary changes
 #
-# Usage: make_voxceleb1.pl /export/voxceleb1 data/
+# Usage: make_voxceleb1_v1.pl /export/voxceleb1 dev data/dev
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v1/local/make_voxceleb1.pl]: Make sure your usage messages are correct.

File: egs/voxceleb/v1/local/make_voxceleb1.pl [hunk 5]
```diff
@@ -1,130 +1,127 @@
 #!/usr/bin/perl
 #
-# Copyright 2018  Ewald Enzinger
-#           2018  David Snyder
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v1/local/make_voxceleb1.pl]: You can put your own copyright in the new version of the file (make_voxceleb_v2.pl), if you want.

File: egs/voxceleb/v1/local/make_voxceleb1.pl [hunk 6]
```diff
@@ -1,130 +1,127 @@
...
+# Copied from make_voxceleb2.pl and made necessary changes
 #
-# Usage: make_voxceleb1.pl /export/voxceleb1 data/
+# Usage: make_voxceleb1_v1.pl /export/voxceleb1 dev data/dev
+#
+# Note: This script requires ffmpeg to be installed and its location included in $PATH.
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v1/local/make_voxceleb1.pl]: Is ffmpeg actually required? It looks like you commented out some code related to that.

File: egs/voxceleb/v1/local/make_voxceleb1.pl [hunk 7]
```diff
@@ -1,130 +1,127 @@
...
   exit(1);
 }
 
-($data_base, $out_dir) = @ARGV;
-my $out_test_dir = "$out_dir/voxceleb1_test";
-my $out_train_dir = "$out_dir/voxceleb1_train";
+# Check that ffmpeg is installed.
+#if (`which ffmpeg` eq "") {
+# die "Error: this script requires that ffmpeg is installed.";
+#}
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v1/local/make_voxceleb1.pl]: If ffmpeg isn't required, this commented out block of code should be removed.

File: egs/voxceleb/v1/local/make_voxceleb1.pl [hunk 8]
```diff
@@ -1,130 +1,127 @@
...
+
+		        	my $target = "nontarget";
+					 if ($tar_or_non eq "1") {
+						$target = "target";
+							    }
+			        print TRIAL_OUT "$utt_id1 $utt_id2 $target\n";
+			      }
+			      close(TRIAL_OUT) or die;
+			      close(TRIAL_IN) or die;
+			      close(META_IN) or die;
```
Comment (david-ryan-snyder) [inline egs/voxceleb/v1/local/make_voxceleb1.pl]: Please consistently use 2 spaces for indentation throughout the script.

**Comment by sayint-ai** [comment]
Sure will do.

**Comment by david-ryan-snyder** [comment]
OK, let me know when you're done and I'll review it again.

BTW, there's no need make a new run.sh file for this and there's no need to modify the cmd.sh. Also, the original make_voxceleb1.pl file should be left unchanged, there shouldn't be any diff for that file. The only changes that should be in this PR are the addition of a new file called make_voxceleb1_v2.pl, and a few lines changed in the run.sh file.

**Comment by sayint-ai** [comment]
The default train_cmd uses --mem 4G for even stuff like mfcc extraction. This shouldn't be required for stuff like feature extraction especially on nodes where the NCPU to memory ratio could be less than 4G. I'm seeing lesser CPU utiilization with the --mem 4G option on feature extraction. RAM utlization doesn't seem to be. Hence the changes to cmd.sh. I'm running the scripts and will let you know when all the stages run through successfull. We did notice a EER of around 2.8% in the initial run against 3.128% in the comments in run.sh. Just want to recheck that as well in case there was something wrong in our scripts that showed a lower EER.

**Comment by danpovey** [comment]
It looks like this has already been addressed some other way, so closing this PR

**Pull request closed without merging**